watcher.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	"github.com/fsnotify/fsnotify"
+	pollingWatcher "github.com/radovskyb/watcher"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"syscall"
+	"time"
+)
+
+type FileWatcher interface {
+	Close() error
+	AddFiles(pattern *regexp.Regexp) error
+	add(path string) error
+	Watch(jobs chan<- string, pattern *regexp.Regexp)
+}
+
+type NotifyWatcher struct {
+	watcher *fsnotify.Watcher
+}
+
+func (n NotifyWatcher) Close() error {
+	return n.watcher.Close()
+}
+
+func (n NotifyWatcher) AddFiles(pattern *regexp.Regexp) error {
+	return addFiles(n)
+}
+
+func (n NotifyWatcher) Watch(jobs chan<- string, pattern *regexp.Regexp) {
+	for {
+		select {
+		case ev := <-n.watcher.Events:
+			if ev.Op&fsnotify.Remove == fsnotify.Remove || ev.Op&fsnotify.Write == fsnotify.Write || ev.Op&fsnotify.Create == fsnotify.Create {
+				base := filepath.Base(ev.Name)
+
+				// Assume it is a directory and track it.
+				if *flagRecursive == true && !flagExcludedDirs.Matches(ev.Name) {
```

Comment:
this is duplicated code (in PollingWatcher). Can you refactor this into something like `directoryShouldBeTracked(ev.Name)`?

Reply:
sure :+1: